packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx

```diff
@@ -204,6 +227,162 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
+      });
+    } else {
+      // Add rounded corners to first and last highlighted squares in the last week
+      weekCorners.push({
+        index: 0,
+        styles: styles.roundedCornersBottomLeft,
+        week: numberOfWeeks - 1
+      },
+        {
+          index: lastWeek[lastWeek.length - 1],
+          styles: styles.roundedCornersBottomRight,
+          week: numberOfWeeks - 1
+        });
+    }
+
+    // If the last week does not have the whole week highlighted
+    // The last square in the week before it will need rounded corners
+    if (daysInWeek !== lastWeek.length) {
+      weekCorners.push({
+        index: daysInWeek - 1,
+        styles: styles.roundedCornersBottomRight,
+        week: numberOfWeeks - 2
+      });
+    }
+    return weekCorners;
+
+  }
+
+  private findUnFocusedWeekCorners(weeks: any) {
+    let weekCorners: any[] = [];
+    let daysInWeek = weeks[0].length;
+    let numberOfWeeks = weeks.length;
+
+    // Highlighted days in first week of month
+    let firstWeek: any[] = [];
+    weeks[0].map((day: any, index: number) => {
+      if (!day.isInMonth) { firstWeek.push(index); }
+    });
+
+    // Highlighted days in last week of month
+    let lastWeek: any[] = [];
+    weeks[numberOfWeeks - 1].map((day: any, index: number) => {
+      if (!day.isInMonth) { lastWeek.push(index); }
+    });
+
+    if (firstWeek.length === 1) {
+      // If only one highlighted day this square has rounded corners on all sides
+      weekCorners.push({
+        index: 0,
+        styles: styles.roundedCornersFull,
+        week: 0
+      });
+    } else {
+      // Add rounded corners to first and last highlighted squares in the first week
+      weekCorners.push({
+        index: 0,
+        styles: styles.roundedCornersLeft,
+        week: 0
+      },
+        {
+          index: firstWeek[firstWeek.length - 1],
+          styles: styles.roundedCornersRight,
+          week: 0
+        });
+    }
+
+    if (lastWeek.length === 1) {
+      // If only one highlighted day this square has rounded corners on all sides
+      weekCorners.push({
+        index: daysInWeek - 1,
+        styles: styles.roundedCornersFull,
+        week: numberOfWeeks - 1
+      });
+    } else {
+      // Add rounded corners to first and last highlighted squares in the last week
+      weekCorners.push({
+        index: lastWeek[0],
+        styles: styles.roundedCornersLeft,
+        week: numberOfWeeks - 1
+      },
+        {
+          index: daysInWeek - 1,
+          styles: styles.roundedCornersRight,
+          week: numberOfWeeks - 1
+        });
+    }
+    return weekCorners;
+  }
+
+  private checkHighlightCorner(weekCorners: any, dayIndex: number, weekIndex: number) {
+    let cornerStyle = '0px';
+    if (weekCorners) {
+      weekCorners.forEach((corner: any) => {
+        if (corner.index === dayIndex && corner.week === weekIndex) {
+          cornerStyle = corner.styles;
+        }
+      });
+    }
+
+    return cornerStyle;
```

Comment:
This function is actually returning a style so the function should probably be renamed to getSelectedCornerStyle(). Also, instead of looping, it might be more efficient to use a dictionary for your weekCorners, and you can key off of a combination of day and week, i.e., key: "week_day".

Reply:
Yes!  Great idea!